### PR TITLE
Use npm-run instead of shelljs-nodecli

### LINKED ISSUE #3683: Loading "Gruntfile.js" tasks...ERROR
screen shot 2016-10-14 at 16 31 09

I tried to install the "conventional-changelog" but still not work.

### PR DESCRIPTION
**Pull request by misteroneill**
## Description
Fixes #3683 

## Specific Changes proposed
Stop using shelljs-nodecli and start using npm-run! The reason this fails is that shelljs-nodecli makes the assumption that the package and its executable are the same name (e.g. the `grunt` command is exposed by the `grunt` package).

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

**Comment by gkatsev** [comment]
I guess this is fine? I've never had problems with shelljs-nodecli before. And it's been working fine for me, so, not sure exactly what the problem was.

**Comment by misteroneill** [comment]
The problem is that the package name and the CLI name don't match. shelljs-nodecli assumes they do; so, when you look for `conventional-changelog` it looks for `node_modules/conventional-changelog/package.json` and fails to find it and gives up.

**Comment by BrandonOCasey** [comment]
is there any reason not to switch to vanilla `shelljs` and use `shelljs.exec('node_modules/.bin/*')`?

**Comment by misteroneill** [comment]
I don't think that would work on Windows and I believe there are cases where `node_modules/.bin` isn't the right path? Hence the existence of the `npm bin` command.

**Comment by BrandonOCasey** [comment]
LGTM

**Comment by gkatsev** [comment]
fwiw, shelljs-nodecli *is* picking up the local conventional-changelog bin for me...

**Comment by gkatsev** [comment]
Or maybe not.

File: build/grunt.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ module.exports = function(grunt) {
       release: {
         tag_name: 'v'+ version.full,
         name: version.full,
-        body: nodeCli.exec('conventional-changelog', '-p videojs', {silent: true}).output
+        body: npmRun.execSync('conventional-changelog -p videojs', {silent: true})
```
Comment (gkatsev) [inline build/grunt.js]: This needs an `encoding: 'utf8'` options because a buffer is returned by default.

**Pull request merged**